tony-core/src/main/java/com/linkedin/tony/util/Utils.java

```diff
@@ -351,10 +352,17 @@ public static void addEnvironmentForResource(LocalResource resource, FileSystem
    * @param conf the TonY configuration.
    * @return map from configured job name to its corresponding resource request
    */
-  public static Map<String, TensorFlowContainerRequest> parseContainerRequests(Configuration conf) {
+  public static Map<String, JobContainerRequest> parseContainerRequests(Configuration conf) {
     Set<String> jobNames = getAllJobTypes(conf);
-    Map<String, TensorFlowContainerRequest> containerRequests = new HashMap<>();
+    Set<String> untrackedJobTypes = Arrays.stream(getUntrackedJobTypes(conf)).collect(Collectors.toSet());
+    Map<String, JobContainerRequest> containerRequests = new HashMap<>();
     int priority = 0;
+
+    List<String> prepareStageTasks = new ArrayList<>(conf.getTrimmedStringCollection(TonyConfigurationKeys.APPLICATION_PREPARE_STAGE));
+    List<String> trainingStageTasks = new ArrayList<>(conf.getTrimmedStringCollection(TonyConfigurationKeys.APPLICATION_TRAINING_STAGE));
+    ensureStagedTasksIntegrity(prepareStageTasks, trainingStageTasks, jobNames);
+    List<String> tasksToDependOn = prepareStageTasks.stream().filter(x -> !untrackedJobTypes.contains(x)).collect(Collectors.toList());
```

Comment:
I guess the intent here is we don't want to depend on tasks which will never terminate. Is this the api we want to expose though? Ithink it's a bit confusing here, the user would need to specify "dependent" jobs (both terminating and non-terminating), then specify the non-terminating one in non-tracked job types.

Reply:
I think the api we expose is prepare-stage and training-stage, there's no notion of what task depend on what, but rather what tasks are launched in what stage. Once we open up the dependsOn api, these code can be removed.